src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotationMap.java

```diff
@@ -229,4 +235,111 @@ public static FuncotationMap createAsAllTableFuncotationsFromVcf(final String tr
     private static boolean areDuplicateTranscriptIDsFound(final List<GencodeFuncotation> gencodeFuncotations) {
         return gencodeFuncotations.size() != new HashSet<>(gencodeFuncotations).size();
     }
+
+    /**
+     * Get all field names found in the funcotations for the given transcript ID.
+     *
+     * @param transcriptId transcript ID.  Never {@code null}
+     * @return Field names in the funcotations.  Empty set if no funcotations (or funcotations have no fields).
+     * Never {@code null}
+     */
+    public Set<String> getFieldNames(final String transcriptId) {
+        Utils.nonNull(transcriptId);
+
+        final LinkedHashSet<Funcotation> funcotations =  txToFuncotations.getOrDefault(transcriptId, new LinkedHashSet<>());
+        return funcotations.stream().map(Funcotation::getFieldNames).flatMap(LinkedHashSet::stream).collect(Collectors.toSet());
+    }
+
+    /**
+     * See {@link FuncotationMap#getFieldNames(String)}, but this returns field names for all transcripts and all alleles.
+     * @return  See {@link FuncotationMap#getFieldNames(String)}
+     */
+    private Set<String> getFieldNames() {
+        final List<String> txIds = getTranscriptList();
+        final Set<String> result = new HashSet<>();
+        txIds.forEach(txId -> result.addAll(getFieldNames(txId)));
+        return result;
+    }
+
+    /**
+     * See {@link FuncotationMap#getFieldNames(String)}, but this returns field names for a single transcripts and a
+     *  single allele.
+     * @param transcriptId See {@link FuncotationMap#getFieldNames(String)}
+     * @param allele Never {@code null}
+     * @return See {@link FuncotationMap#getFieldNames(String)}
+     */
+    private Set<String> getFieldNames(final String transcriptId, final Allele allele) {
+        Utils.nonNull(transcriptId);
+        Utils.nonNull(allele);
+
+        final LinkedHashSet<Funcotation> funcotations =  txToFuncotations.getOrDefault(transcriptId, new LinkedHashSet<>());
+        return funcotations.stream()
+                .filter(f -> f.getAltAllele().equals(allele))
+                .map(Funcotation::getFieldNames)
+                .flatMap(LinkedHashSet::stream).collect(Collectors.toSet());
```

Comment:
I would argue that there is no order to preserve once you call this.  However, just in case it becomes important later, I have altered the code to preserve whatever order is there.

Done.